extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeDashManifestCreator.java

```diff
@@ -0,0 +1,1614 @@
+        } catch (final DOMException e) {
+            throw new YoutubeDashManifestCreationException(
+                    "Could not generate or append to the document the SegmentTimeline element of the DASH manifest", e);
+        }
+    }
+
+    /**
+     * Generate segment elements for OTF streams.
+     *
+     * <p>
+     * By parsing by the first media sequence, we know how many durations and repetitions they are
+     * so we just have to loop into {@link #SEGMENTS_DURATION} and {@link #DURATION_REPETITIONS}
+     * to generate for each duration, the following element:
+     * <br>
+     * {@code <S d="segmentDuration" r="durationRepetition" />}
+     * <br>
+     * These elements will be appended as children of the {@code <SegmentTimeline>} element.
+     * </p>
+     * <p>
+     * The {@code <SegmentTimeline>} element needs to be generated before this element with
+     * {@link #generateSegmentTimelineElement(Document)}.
+     * </p>
+     *
+     * @param document the {@link Document} on which the the {@code <S>} elements will be appended
+     * @throws YoutubeDashManifestCreationException if something goes wrong when generating or
+     *                                              appending the {@code <S>} elements to the
+     *                                              document
+     */
+    private static void generateSegmentElementsForOtfStreams(@Nonnull final Document document)
+            throws YoutubeDashManifestCreationException {
+        try {
+            if (isNullOrEmpty(SEGMENTS_DURATION) || isNullOrEmpty(DURATION_REPETITIONS)) {
+                throw new IllegalStateException(
+                        "Duration of segments and/or repetition(s) of segments are unknown");
+            }
+            final Element segmentTimelineElement = (Element) document.getElementsByTagName(
+                    "SegmentTimeline").item(0);
+
+            for (int i = 0; i < SEGMENTS_DURATION.size(); i++) {
+                final Element sElement = document.createElement("S");
+
+                final int durationRepetition = DURATION_REPETITIONS.get(i);
+                if (durationRepetition != 0) {
+                    final Attr rAttribute = document.createAttribute("r");
+                    rAttribute.setValue(String.valueOf(durationRepetition));
+                    sElement.setAttributeNode(rAttribute);
+                }
+
+                final Attr dAttribute = document.createAttribute("d");
+                dAttribute.setValue(String.valueOf(SEGMENTS_DURATION.get(i)));
+                sElement.setAttributeNode(dAttribute);
+
+                segmentTimelineElement.appendChild(sElement);
+            }
+
+        } catch (final DOMException | IllegalStateException | IndexOutOfBoundsException e) {
+            throw new YoutubeDashManifestCreationException(
+                    "Could not generate or append to the document the Segment elements of the DASH manifest", e);
```

Comment:
That order should be changed here and when creating similar `YoutubeDashManifestCreationException`s: 
```suggestion
                    "Could not generate or append the Segment elements of the DASH manifest  to the document", e);
```